Add testing task 2

tsbonev opened the pull request.

mgenov commented on src/com/clouway/crm/core/ArrayHandler.java:
```diff
@@ -0,0 +1,126 @@
+
+public class ArrayHandler {
+
+    private int[] returnArray; //handler's own copy of an array
```
Could be defined in the method as keep it here may cause memory leaks in some situations and it's better variables to be with shorted possible scope as possible.

mgenov commented on src/com/clouway/learning/introducingjava/Main.java:
```diff
@@ -0,0 +1,100 @@
+     * @param b second number to be calculated
+     * @return the largest common divisor
+     */
+    public static int CommonDivisor(int a, int b){
```
This looks like not belongs here. Also method names in java should start with lower case character.

mgenov commented on tests/com/clouway/crm/core/ArrayHandlerTest.java:
```diff
@@ -0,0 +1,45 @@
+    @Test
+    void testPrintArray() {
+        int[] testArr = {1, 2, 3};
+        String testStr = "123";
```
Strange output. More reasonable should be [1, 2, 3] instead of "123" ...

mgenov commented on tests/com/clouway/crm/core/ArrayHandlerTest.java:
```diff
@@ -0,0 +1,45 @@
+    void testReverseArray() {
+        int[] testArr = {1, 2, 3};
+        int[] testReverse = {3, 2, 1};
+        assertArrayEquals(testArr, handler.reverseArray(testReverse));
```
Could be simplified with something like: 
```java
assertArrayEquals(handler.reverseArray(new int[] {1,2,3}), new int[] {3,2,1})
```

Note that the "test" part in the name doesn't give any value or meaning. 

Keep in mind that

mgenov commented on tests/com/clouway/crm/core/ArrayHandlerTest.java:
```diff
@@ -0,0 +1,45 @@
+    static ArrayHandler handler = new ArrayHandler();
+
+    @Test
+    void testReverseArray() {
```
Junit 4 doesn't require method names to start with test keyword. Just:

```java
@Test
public void reverseArray() {
  ....
}

```

mgenov commented on tests/com/clouway/crm/core/ArrayHandlerTest.java:
```diff
@@ -0,0 +1,45 @@
+
+class ArrayHandlerTest {
+
+    static ArrayHandler handler = new ArrayHandler();
```
static ? Looks strange to be defined with static as it will ensure that single instance will be used.

tsbonev commented:
Removed unnecessary code from Main
Moved returnArray into the parameters of the methods that use it
Renamed the test methods to exclude the test keyword
Removed the static from the ArrayHandler field
Simplified the parameters for the reverse test
Changed the output of the printArray to match the normal .toString() format
